### ISSUE TITLE
Clarify required React version?

### ISSUE THREAD

**Issue by veltman** [material-components/material-components-web-react#640]
I ran into some inscrutable `ref` errors trying to use these components on a project running React 16.2, and eventually realized that it relies on the `createRef()` API added in 16.3.

Assuming this is true across the board, it might be worth adding a note to the README that 16.3+ is required, or listing it as a peerDependency? (I can submit a PR if you can confirm that 16.3 is the intended min version, not sure whether there are other behaviors in later versions that it also relies on)

**Comment by moog16** [comment]
We also use `React.Fragment` is another feature that we have, but it looks like it was added in [16.2](https://github.com/facebook/react/blob/master/CHANGELOG.md#1641-june-13-2018). I took a look at the [React Changelog](https://github.com/facebook/react/blob/master/CHANGELOG.md) of all the most recent versions since 16.3, and it looks like we don't use of those features. So `v16.3+` is correct.

I would definitely appreciate a PR!

**Comment by veltman** [comment]
Thinking about this a bit more it seems like it might be preferable to make the min version 16.4.2, since that's what's currently being tested against - or to roll back the devDependencies to 16.3.0 - what do you think?

**Comment by moog16** [comment]
I think the earlier the version the wider the audience that can use the components. So my inclination is to rollback devDeps to 16.3.0. But like you said we've been using 16.4.2 as what has been tested against.  I think making the min version 16.4.2 is the safer bet.

**Comment by veltman** [comment]
Sounds good - it also wouldn't prevent people from using an earlier version, just warn.

**Comment by moog16** [comment]
Right, but they wouldn't be able to anyways with the different features we use in the repo. Or it would force them to update. So this at least will throw them a warning instead of them having sift through the changelog and figuring out why its not working (like you had to do).
